File: unity/Assets/Scripts/Multiplayer/Matchmaking/Requests/MatchmakingJoinRequestResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Matchmaking.Requests
{
    /// <summary>
    /// Response returned when a player requests to join a matchmaking lobby.
    /// Contains the request ID for tracking.
    /// </summary>
    [System.Serializable]
    public class MatchmakingJoinRequestResponse : ApiResponse<EMatchmakingJoinError>
    {
        /// <summary>
        /// The unique ID of the join request.
        /// </summary>
        public string request_id;

        /// <summary>
        /// Confirmation message for the join request.
        /// </summary>
        public string message;
    }
}

File: java/src/main/java/com/michitai/multiplayer/matchmaking/requests/MatchmakingJoinRequestResponse.java
package com.michitai.multiplayer.matchmaking.requests;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

/**
 * Response returned when a player requests to join a matchmaking lobby.
 */
public class MatchmakingJoinRequestResponse extends ApiResponse {
    @JsonProperty("message")
    private String message;

    @JsonProperty("request_id")
    private String requestId;

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getRequestId() {
        return requestId;
    }

    public void setRequestId(String requestId) {
        this.requestId = requestId;
    }
}
